packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -500,8 +500,8 @@ export class SchemaFactory<
 			return undefined;
 		});
 		return createFieldSchema(FieldKind.Optional, t, {
-			...props,
 			defaultProvider: defaultOptionalProvider,
+			...props,
```

Comment:
@daesunp - Is this appropriate? Before, it was impossible to ever pass in a provider because it was always overwritten by the default one. I think it makes more sense to have a given provider take precedence over the default. We don't let users pass providers in the public API regardless, so it doesn't currently matter to them, but I cheat and force a provider in in my unit test - which needs this change to work.